great_ape_safe/ape_api/uni_v2.py

```diff
@@ -145,7 +146,7 @@ def swap_exact_eth_for_tokens(self, amountIn, path, destination=None):
         # https://docs.uniswap.org/protocol/V2/reference/smart-contracts/router-02#swapethforexacttokens
         destination = self.safe.address if not destination else destination
 
-        tokenOut = path[-1]
+        tokenOut = path[0][-2] if type(path[0][-1]) == bool else path[-1]
```

Comment:
suggesting to test for `len(path)` instead of `type(path[0][-1])` (or `type(path[0] == tuple`?); bit weird to subscript `path[0]` in the traditional case where it is an actual address.

Reply:
that can be reverted back to `tokenOut = path[-1]` now that correct path is built inside the function